### ISSUE TITLE
Not working with Godot 4.2.1

### ISSUE THREAD

**Issue by devantti** [Federico-Ciuffardi/GodotTouchInputManager#37]
I updated my Gogot engine from 4.1.3 to 4.2.1 and touches stopped to work on mobile. When I downgraded back to 4.1.3 it is working now again.

**Comment by Federico-Ciuffardi** [comment]
I tried https://github.com/Federico-Ciuffardi/GodotTouchInputManager-Demo and https://github.com/Federico-Ciuffardi/GestureControlledCamera2D in version `v4.2.1.stable` and they seem to work fine.

I suspect the problem is related to the way you use GDTIM in your project. Could you send a minimal reproducible example so I can try to figure out what is wrong?

**Comment by devantti** [comment]
I will send it later. Did you tried it on emulator or physical device?

**Comment by devantti** [comment]
I wasn't sure how to do it so here you can download a zip file: https://drive.google.com/file/d/1FkkgnIeiHxdENQbS8I2WR8tDzbL0R_Nv/view?usp=sharing.

In Godot 4.2.1 I created a new project from asset library projects (Godot touch input manager demo  - Godot 4.x). To get it to emulator I installed android build template from project menu. Then when emulator (Pixel 7 Pro API 33) is running pushed remote debug in editor to get it to emulator. It crashed right after starting so changed stretch mode to viewport and renderer to gl_compatibility/gl_compatibility. When I tried it it didn't register any touch on emulator. Then tried again with Godot 4.1.3 and everything works as expected.

**Comment by Federico-Ciuffardi** [comment]
I tried your project with a physical device and it worked, so maybe the problem is related to the emulator or your specific development environment (OS, Android SDK, etc). 

If possible, try with a physical device to verify that the emulator is not the problem.

Another thing you could check is if Godot native touch events ([InputEventScreenTouch](https://docs.godotengine.org/en/stable/classes/class_inputeventscreentouch.html) and [InputEventScreenDrag](https://docs.godotengine.org/en/stable/classes/class_inputeventscreendrag.html#class-inputeventscreendrag)) work on the emulator.

**Comment by devantti** [comment]
Interesting... Like I told earlier I have also tried my own project on physical device and there it didn't work. I will try demo on device too when I get home from work.

I was thinking about moving away from your library and just use those events but maybe I should first try them too. :)

**Comment by devantti** [comment]
I made a simple project from scratch. Touch (InputEventScreenDrag) works on 4.1.3 but not in 4.2.1 on emulator not device. For some reason on 4.2.1 on emulator and device touch is registered as mouse left and drag as mouse motion regardless is emulate mouse from touch is on or off.

`extends Node2D

@onready var label = $Label

func _input(event: InputEvent):
	label.text = event.as_text()
	if event is InputEventScreenTouch:
		var touch_position = event.position
		var label_node = Label.new()
		label_node.text = "Touch"
		label_node.position = touch_position
		add_child(label_node)

	elif event is InputEventScreenDrag:
		var touch_position = event.position
		var label_node = Label.new()
		label_node.text = "Drag"
		label_node.position = touch_position
		add_child(label_node)`

Thanks for the help. I should get new mac next week so maybe fresh install will iron out this problem. Thanks for the help!

**Comment by Patataonk** [comment]
Hi
I have the same issue

downloaded the demo https://github.com/Federico-Ciuffardi/GodotTouchInputManager-Demo
exported as apk and tried on a android phone and it doesnt work

My OS is windows

**Comment by devantti** [comment]
Nope, that didn't help. Still same issue on mac with 4.2.1 Stable.

**Comment by devantti** [comment]
Problem is fixed in Godot 4.3 beta 1.
